app/features-json/json_authenticated_controller_base.rb

```diff
@@ -0,0 +1,44 @@
+require_relative "../shared/controller_base"
+require_relative "../shared/logging_module"
+require_relative "../services/config_service"
+require_relative "../services/services"
+require_relative "./middleware/jwt_auth"
+
+module FastlaneCI
+  #
+  # A user is required to be logged in to access any routes on this controller
+  #
+  class JSONAuthenticatedControllerBase < ControllerBase
+    use FastlaneCI::JwtAuth
+    include FastlaneCI::Logging
+
+    # Method to obtain the current user given their primary key.
+    # Raises if no User found.
+    # @return [User]
+    def user!
```

Comment:
@minuscorp for these kinds of find-or-raise patterns, I think we should start making use of Sinatra's `halt` method. As described here: http://myronmars.to/n/dev-blog/2012/01/why-sinatras-halt-is-awesome

This way, it's not raising an exception that must be caught, and this doesn't need to be a bang method.

Also, could we call it `current_user` :)